packages/@aws-cdk-testing/framework-integ/test/aws-ecs/test/integ.cluster-windows-server-ami.ts

```diff
@@ -0,0 +1,57 @@
+import * as autoscaling from 'aws-cdk-lib/aws-autoscaling';
+import * as ec2 from 'aws-cdk-lib/aws-ec2';
+import * as iam from 'aws-cdk-lib/aws-iam';
+import * as cdk from 'aws-cdk-lib';
+import * as ecs from 'aws-cdk-lib/aws-ecs';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+
+const app = new cdk.App();
+const stack = new cdk.Stack(app, 'integ-ecs-windows-server-ami');
+
+const vpc = new ec2.Vpc(stack, 'Vpc', { maxAzs: 1 });
+const cluster = new ecs.Cluster(stack, 'Cluster', {
+  vpc,
+});
+
+const insRole = new iam.Role(stack, 'InstanceRole', {
+  assumedBy: new iam.ServicePrincipal('ec2.amazonaws.com'),
+  managedPolicies: [
+    iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonEC2ContainerServiceForEC2Role'), // for ECS
+    iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSSMManagedInstanceCore'), // for SSM
+  ],
+});
+insRole.applyRemovalPolicy(cdk.RemovalPolicy.DESTROY);
+
+const autoScalingGroup = new autoscaling.AutoScalingGroup(stack, 'ASG', {
+  vpc,
+  role: insRole,
+  instanceType: ec2.InstanceType.of(ec2.InstanceClass.T2, ec2.InstanceSize.LARGE),
+  machineImage: ecs.EcsOptimizedImage.windows(ecs.WindowsOptimizedVersion.SERVER_2022),
+  minCapacity: 1,
+});
+
+const cp = new ecs.AsgCapacityProvider(stack, 'EC2CapacityProvider', {
+  autoScalingGroup,
+  enableManagedTerminationProtection: false,
+});
+
+cluster.addAsgCapacityProvider(cp);
+
+const taskDefinition = new ecs.Ec2TaskDefinition(stack, 'TaskDef', {});
+
+taskDefinition.addContainer('fluent-bit', {
+  image: ecs.ContainerImage.fromRegistry('public.ecr.aws/aws-observability/aws-for-fluent-bit:windowsservercore-stable'),
+  memoryLimitMiB: 2096,
+  cpu: 1024,
+});
```

Comment:
May be it would be better also to have an assert to make sure we are able to call the containerized application. I think we can have this as separate PR later. What do you think?

Reply:
Do you mean calling the containerized application from the outside? If so, the difficulty there is that I couldn't find an image in the ECR registry that 1. expose an endpoint and 2. runs on Windows.